### ISSUE TITLE
default_scope breaks chained having statements in rails4

### ISSUE THREAD

**Issue by helloenvoy** [rails/rails#13965]
A model with default_scope appears not to accept additional having statements.  Here's an example:

```
> rails g migration create_test_model
> rake db:migrate

class TestModel < ActiveRecord::Base; end

> rails c
> TestModel.having('x').having('y').to_sql
 => "SELECT `test_models`.* FROM `test_models`  HAVING x AND y" 

class TestModel < ActiveRecord::Base
  default_scope -> { order('z') } # add a default scope
end

> rails c
> TestModel.having('x').having('y').to_sql
 => "SELECT `test_models`.* FROM `test_models`  HAVING x  ORDER BY z" 
```

This was a regression somewhere between 3.2.16 and 4.0.2.

**Comment by htanata** [comment]
This seems to be fixed on master now (as of fcbd2e8). This test passes for me:

```diff
diff --git a/activerecord/test/cases/scoping/default_scoping_test.rb b/activerecord/test/cases/scoping/default_scoping_test.rb
index 0dbc609..ce295a6 100644
--- a/activerecord/test/cases/scoping/default_scoping_test.rb
+++ b/activerecord/test/cases/scoping/default_scoping_test.rb
@@ -453,4 +453,11 @@ class DefaultScopingTest < ActiveRecord::TestCase
     assert_equal 1, scope.where_clause.ast.children.length
     assert_equal Developer.where(name: "David"), scope
   end
+
+  test "default scope with havings" do
+    havings = -> (klass) { klass.having('sum(id) = 0').having('sum(id) = 0') }
+
+    assert_equal havings.call(Developer.where(name: "Jamis")),
+      havings.call(DeveloperCalledJamis)
+  end
 end
```

If needed, I can do a PR with this test to prevent regression.

**Comment by kamipo** [comment]
Closing since this has not happened at least from 4.2.
